src/Pgvector/Npgsql/VectorConverter.cs

```diff
@@ -57,16 +57,16 @@ public override void Write(PgWriter writer, Vector value)
         if (writer.ShouldFlush(sizeof(ushort) * 2))
             writer.Flush();
 
-        var vec = value.ToArray();
-        var dim = vec.Length;
+        var span = value.Memory.Span;
```

Comment:
This accesses the data directly, removing an additional memory allocation and copy.